tests/async/test_browsercontext_add_cookies.py

```diff
@@ -0,0 +1,387 @@
+# Copyright (c) Microsoft Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import datetime
+
+import pytest
+
+from playwright import Error
+
+
+async def test_should_work(context, page, server):
+    await page.goto(server.EMPTY_PAGE)
+    await context.addCookies(
+        [{"url": server.EMPTY_PAGE, "name": "password", "value": "123456"}]
+    )
+    assert await page.evaluate("() => document.cookie") == "password=123456"
+
+
+async def test_should_roundtrip_cookie(context, page, server):
+    await page.goto(server.EMPTY_PAGE)
+    # @see https://en.wikipedia.org/wiki/Year_2038_problem
+    date = int(datetime.datetime(2038, 1, 1).timestamp() * 1000)
```

Comment:
I manually checked this matches the date from the JS version, but someone should double check since dates are always fun!